torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -222,14 +223,20 @@ void DistEngine::executeSendFunction(
 
     // Run the autograd engine.
     runEngineAndAccumulateGradients(autogradContext, dummyRoot, outputEdges);
-
-    // Wait for all of the outstanding rpcs to complete.
-    autogradContext->clearAndWaitForOutstandingRpcs();
+    guard.cancel();
```

Comment:
This guard can be removed altogether from the looks of it.

Reply:
Yes, I agree.